Automattic/jetpack
# Contact Form: disable buttons when the form is submitted

Fixes #18708

#### Changes proposed in this Pull Request:
Currently, the contact form can be clicked multiple times, resulting in multiple form submissions. Fix this by disabling the submitted form's button elements when a form is submitted.

* Create a new method, `get_script_for_form`, which adds an embedded script to the contact form html.
* In the script, disable the form's button elements when an element with the `contact-form` class is submitted.

Alternatives Considered
* I considered creating a new external file for the script, but the script is small and focused so an embedded script seemed fine. (It appears the contact form doesn't already load an external JS file.)

#### Jetpack product discussion
* n/a

#### Does this pull request change what data or activity we track or use?
* no

#### Testing instructions:
1. Install, activate, and connect this branch of Jetpack.
2. In wp-admin, create a new post and add a Contact Form block to the post.
3. Publish the new post.
4. Navigate to the new post. Fill out the contact form and submit it. The contact form should be submitted successfully.
5. In wp-admin, navigate to the Feedback menu item and confirm that the contact form submission was received.
6. Repeat steps 4-5. This time, attempt to click the Submit button multiple times.
7. In wp-admin, navigate to the Feedback menu item and confirm that the contact form submission was received once.
8. Repeat steps 4-5. This time, attempt to submit the form multiple times by tabbing to the Submit button and clicking 'enter' multiple times.
9. In wp-admin, navigate to the Feedback menu item and confirm that the contact form submission was received once.
10. Repeat steps 4-9 in different browsers (Chrome, Firefox, Safari).
11. Navigate to the new post on a mobile device. Fill out the contact form and attempt to click the submit button multiple times. The contact form should be submitted successfully.
12. In wp-admin, navigate to the Feedback menu item and confirm that the contact form submission was received once.
